corehq/apps/performance_sms/views.py

```diff
@@ -1,57 +1,114 @@
 from django.contrib import messages
 from django.core.urlresolvers import reverse
 from django.http import HttpResponseRedirect
-from django.shortcuts import render
+from django.utils.decorators import method_decorator
 from django.utils.safestring import mark_safe
-from django.utils.translation import ugettext as _
+from django.utils.translation import ugettext as _, ugettext_lazy
 from django.views.decorators.http import require_POST
 from corehq import toggles
+from corehq.apps.domain.views import BaseDomainView
 from corehq.apps.performance_sms import dbaccessors
 from corehq.apps.performance_sms.forms import PerformanceMessageEditForm
 from corehq.apps.performance_sms.message_sender import send_messages_for_config
 from corehq.apps.performance_sms.models import PerformanceConfiguration
 from corehq.apps.reminders.views import reminders_framework_permission
+from corehq.apps.style.decorators import use_bootstrap3
 from corehq.util import get_document_or_404
+from dimagi.utils.decorators.memoized import memoized
 
 
-@reminders_framework_permission
-@toggles.SMS_PERFORMANCE_FEEDBACK.required_decorator()
-def list_performance_configs(request, domain):
-    return render(request, "performance_sms/list_performance_configs.html", {
-        'domain': domain,
-        'performance_configs': dbaccessors.by_domain(domain)
-    })
+class BasePerformanceSMSView(BaseDomainView):
```

Comment:
can you remind me why it's important to convert these to class based views? just looking at the diff it seems like the original code is much simpler and easier to understand whereas there's lots of indirection in the CBV. not saying we should switch it back just want to know why we do this.